### PR TITLE
Add cast for size_t while logging 

### PR DESCRIPTION
**Pull request by pvyawaha**
*Issue #, if available:* #199

*Description of changes:*
This change adds a cast when logging size_t type .

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by ser90368141** [comment]
Hi pvyawaha!
Please add casting in these two places.
https://github.com/aws/ota-for-aws-iot-embedded-sdk/blob/5d7c7f6bdaaef1dddcf0b353d68e01037d7680f6/source/ota.c#L1627-L1630

https://github.com/aws/ota-for-aws-iot-embedded-sdk/blob/5d7c7f6bdaaef1dddcf0b353d68e01037d7680f6/source/ota.c#L2003-L2004
I added these changes and the library built successfully.
Thank!

**Review by yanjos-dev — DISMISSED** [review]

**Review by divekarshubham — APPROVED** [review]

**Review by divekarshubham — APPROVED** [review]

File: source/ota.c [hunk 1]
```diff
@@ -2001,7 +2001,7 @@ static OtaJobParseErr_t handleCustomJob( const char * pJson,
             err = OtaJobParseErrNonConformingJobDoc;
 
             LogDebug( ( "Failed to parse custom job document: OtaJobParseErr_t=%s, jobIdLength=%lu",
-                        OTA_JobParse_strerror( jobDoc.parseErr ), jobDoc.jobIdLength ) );
+                        OTA_JobParse_strerror( jobDoc.parseErr ), ( unsigned long ) jobDoc.jobIdLength ) );
```
Comment (gkwicker) [inline source/ota.c:2004]: See previous comment.

**Review by gkwicker — CHANGES_REQUESTED** [review]
See comments.

File: source/ota.c [hunk 2]
```diff
@@ -1615,7 +1615,7 @@ static DocParseErr_t extractAndStoreArray( const char * pKey,
             LogError( ( "Memory allocation failed "
                         "[key: valueLength]=[%s: %lu]",
                         pKey,
-                        valueLength ) );
+                        ( unsigned long ) valueLength ) );
```
[the remark this answers is not in the document]
Comment (pvyawaha) [reply source/ota.c:1618]: %zu was not used for portability reason with older VS. https://docs.microsoft.com/en-us/previous-versions/tcxf1dw6(v=vs.140)?redirectedfrom=MSDN

**Review by gkwicker — APPROVED** [review]

**Pull request merged**